Dockerfile

```diff
@@ -1,10 +1,16 @@
+# Dockerfile focused on development environment use case
 FROM openjdk:8-jdk-alpine
-ARG usedb=h2
-VOLUME /tmp
-RUN mkdir /work
-COPY . /work
-RUN mv /work/src/main/resources/application.properties.${usedb} /work/src/main/resources/application.properties
-WORKDIR /work
-RUN /work/gradlew build
-RUN mv /work/build/libs/*.jar /work/app.jar
-ENTRYPOINT ["java","-Djava.security.egd=file:/dev/./urandom","-jar","/work/app.jar"]
+
+# Prep image with gradle cache to use as mount point for host cache when container runs
+RUN mkdir /root/.gradle
+
+# We can mount the host code repo here if we to update with every run, but the copy below makes mounting unnecessary
+# if we only want the repo state at build time.
+RUN mkdir /app
+
+WORKDIR /app
+COPY . .
+
+# Running with gradle in dev environment to enable devtools features (debugging, auto reload)
+# This entrypoint is overridden by the Jenkins Pipeline plugin, so we won't bootRun there
+CMD ["./gradlew", "bootRun"]
```

Comment:
This is the magic that starts the app in auto-restart mode (as opposed to java -jar)